src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -493,3 +502,18 @@
       return(list(error = e))
    })
 })
+
+.rs.addFunction("preview_data_import_async", function(jsonData)
+{
+   tryCatch({
+      params = jsonlite::fromJSON(jsonData)
```

Comment:
Yes, I agree that we should take the .RData approach that Jonathan advocates for. We try hard to not have dependencies on R packages in our main implementation (other than the obviously required dependencies on things like shiny, rmarkdown, readr, etc.).